File: legacy/csharp/components/Strata.Application/Interception/IUnitOfWorkPropertySupplier.cs
//  ##########################################################################
//  # File Name: IUnitOfWorkPropertySupplier.cs
//  ##########################################################################

using Strata.Application.Messaging;
using Strata.Domain.UnitOfWork;
using Strata.Foundation.Logging;

namespace Strata.Application.Interception
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// This interface must be implemented by types that are decorated
    /// by the <c>UnitOfWorkInterceptor</c>.
    /// </summary>
    ///  
    public
    interface IUnitOfWorkPropertySupplier
    {
        IUnitOfWorkProvider Provider { get; }
        IActionQueue        Queue { get; }
        ILogger             Logger { get; }
    }
}

//  ##########################################################################

File: java/Components/Strata.Application.Core/src/strata/application/core/interception/IUnitOfWorkPropertySupplier.java
//////////////////////////////////////////////////////////////////////////////
// IUnitOfWorkPropertySupplier.java
//////////////////////////////////////////////////////////////////////////////

package strata.application.core.interception;

import strata.domain.core.unitofwork.IUnitOfWorkProvider;
import strata.foundation.core.action.IActionQueue;

public
interface IUnitOfWorkPropertySupplier
{
    IUnitOfWorkProvider
    getProvider();

    IActionQueue
    getQueue();
}

//////////////////////////////////////////////////////////////////////////////